src/domain/util/app-error.js

```diff
@@ -6,6 +6,10 @@
  * @returns
  */
 export function AppError(error, code = 400, cause = null) {
+  if (error.code > 500) {
```

Comment:
Custom indexes work. Duplicate key mongo errors throw a code that's outside of the HTTP spec